api/tests/mocha/controllers/contact.spec.js

```diff
@@ -0,0 +1,402 @@
+        req = {
+          query: {
+            freetext: invalidFreetext,
+            cursor,
+            limit,
+          }
+        };
+        const err = new InvalidArgumentError(`Invalid freetext: [${invalidFreetext}]`);
+        isOnlineOnly.returns(true);
+        hasAllPermissions.returns(true);
+        contactGetIdsPage.throws(err);
+
+        await controller.v1.getIds(req, res);
+
+        expect(hasAllPermissions.calledOnceWithExactly(userCtx, 'can_view_contacts')).to.be.true;
+        expect(qualifierByContactType.notCalled).to.be.true;
+        expect(qualifierByFreetext.calledOnceWithExactly(req.query.freetext)).to.be.true;
+        expect(dataContextBind.calledOnceWithExactly(Contact.v1.getIdsPage)).to.be.true;
+        expect(contactGetIdsPage.calledOnceWithExactly(freetextOnlyQualifier, cursor, limit)).to.be.true;
+        expect(res.json.notCalled).to.be.true;
+        expect(serverUtilsError.calledOnceWithExactly(err, req, res)).to.be.true;
+      });
+
+      it('returns 400 error when contactType AND freetext is not present', async () => {
+        req = {
+          query: {
+            cursor,
+            limit,
+          }
+        };
+        const err = { status: 400, message: 'Either query param freetext or type is required' };
+        isOnlineOnly.returns(true);
+        hasAllPermissions.returns(true);
+        contactGetIdsPage.throws(err);
+
+        await controller.v1.getIds(req, res);
+
+        expect(hasAllPermissions.calledOnceWithExactly(userCtx, 'can_view_contacts')).to.be.true;
+        expect(qualifierByContactType.notCalled).to.be.true;
+        expect(qualifierByFreetext.notCalled).to.be.true;
+        expect(dataContextBind.notCalled).to.be.true;
+        expect(contactGetIdsPage.notCalled).to.be.true;
+        expect(res.json.notCalled).to.be.true;
+        expect(serverUtilsError.calledOnceWithExactly(err, req, res)).to.be.true;
+      });
+
+      it('rethrows error in case of other errors', async () => {
+        req = {
+          query: {
+            freetext: freetext,
+            type: contactType,
+            cursor,
+            limit,
+          }
+        };
+        const err = new Error('error');
+        isOnlineOnly.returns(true);
+        hasAllPermissions.returns(true);
+        contactGetIdsPage.throws(err);
+
+        await controller.v1.getIds(req, res);
+
+        expect(hasAllPermissions.calledOnceWithExactly(userCtx, 'can_view_contacts')).to.be.true;
+        expect(qualifierByContactType.calledOnceWithExactly(req.query.type)).to.be.true;
+        expect(qualifierByFreetext.calledOnceWithExactly(req.query.freetext)).to.be.true;
+        expect(dataContextBind.calledOnceWithExactly(Contact.v1.getIdsPage)).to.be.true;
+        expect(contactGetIdsPage.calledOnceWithExactly(bothQualifier, cursor, limit)).to.be.true;
+        expect(res.json.notCalled).to.be.true;
+        expect(serverUtilsError.calledOnceWithExactly(err, req, res)).to.be.true;
+      });
```

Comment:
Since we have validation and transformation for `limit` within this part of the code, I think tests are required. If you end up just passing `limit` without transformation, only one test that attests that limit is passed unchanged is sufficient.